app/controllers/usps_shipments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class UspsShipmentsController < ApplicationController
+
+  def estimate
+    if (/\A\d+\z/.match(params[:weight])).nil? || (/\A\d+\z/.match(params[:zip])).nil?
+      render json: { errors: "Please enter a numeric value for weight and zip code."}, status: :bad_request
```

Comment:
I think it's nicer to give the customer a message specific to the field that failed.